shell/platform/linux/fl-view.cc

```diff
@@ -0,0 +1,309 @@
+  }
+}
+
+static gboolean initialize_egl(FlView* self) {
+  /* Note that we don't provide the XDisplay from GTK, this would make both
+   * GTK and EGL share the same X connection and this would crash when used by
+   * a Flutter thread. So the EGL display and GTK both have separate
+   * connections.
+   */
+  self->egl_display = eglGetDisplay(EGL_DEFAULT_DISPLAY);
+
+  EGLint egl_major, egl_minor;
+  if (!eglInitialize(self->egl_display, &egl_major, &egl_minor)) {
+    g_warning("Failed to initialze EGL");
+    return FALSE;
+  }
+  // FIXME: It would probably be useful to store the EGL version for debugging
+  // purposes
+
+  EGLint attributes[] = {EGL_RENDERABLE_TYPE,
+                         EGL_OPENGL_ES2_BIT,
+                         EGL_RED_SIZE,
+                         8,
+                         EGL_GREEN_SIZE,
+                         8,
+                         EGL_BLUE_SIZE,
+                         8,
+                         EGL_ALPHA_SIZE,
+                         8,  // FIXME: Check if Flutter wants alpha?
+                         EGL_NONE};
+  EGLConfig egl_config;
+  EGLint n_config;
+  if (!eglChooseConfig(self->egl_display, attributes, &egl_config, 1,
+                       &n_config)) {
+    g_warning("Failed to choose EGL config");
+    return FALSE;
+  }
+  if (n_config == 0) {
+    g_warning("Failed to find appropriate EGL config");
+    return FALSE;
+  }
+  if (!eglBindAPI(EGL_OPENGL_ES_API)) {
+    g_warning("Failed to bind EGL OpenGL ES API");
+    return FALSE;
+  }
+
+  Window xid = gdk_x11_window_get_xid(gtk_widget_get_window(GTK_WIDGET(self)));
+  self->egl_surface =
+      eglCreateWindowSurface(self->egl_display, egl_config, xid, NULL);
+  EGLint context_attributes[] = {EGL_CONTEXT_CLIENT_VERSION, 2, EGL_NONE};
+  self->egl_context = eglCreateContext(self->egl_display, egl_config,
+                                       EGL_NO_CONTEXT, context_attributes);
+  EGLint value;
+  eglQueryContext(self->egl_display, self->egl_context,
+                  EGL_CONTEXT_CLIENT_VERSION, &value);
+
+  return TRUE;
+}
+
+static void* fl_view_gl_proc_resolver(void* user_data, const char* name) {
+  return reinterpret_cast<void*>(eglGetProcAddress(name));
+}
+
+static bool fl_view_gl_make_current(void* user_data) {
+  FlView* self = static_cast<FlView*>(user_data);
+
+  if (!eglMakeCurrent(self->egl_display, self->egl_surface, self->egl_surface,
+                      self->egl_context))
+    g_warning("Failed to make EGL context current");
+
+  return true;
+}
+
+static bool fl_view_gl_clear_current(void* user_data) {
+  FlView* self = static_cast<FlView*>(user_data);
+
+  if (!eglMakeCurrent(self->egl_display, EGL_NO_SURFACE, EGL_NO_SURFACE,
+                      EGL_NO_CONTEXT))
+    g_warning("Failed to make EGL context current");
+
+  return true;
+}
+
+static uint32_t fl_view_gl_fbo_callback(void* user_data) {
+  /* We only have one frame buffer object - always return that */
```

Comment:
s/We only have/There is only/. "We" is often ambiguous in code comments, so is best avoided.